rtgui/gradient.cc

```diff
@@ -136,88 +136,99 @@ void Gradient::read (const ProcParams* pp, const ParamsEdited* pedited)
+    polCoord1.set(1500.f, float(-degree_ + 180));
+    currLine->begin.setFromPolar(polCoord1);
+    currLine->begin += origin;
+    polCoord1.set(1500.f, float(-degree_    ));
+    currLine->end.setFromPolar  (polCoord1);
+    currLine->end   += origin;
+    currLine = static_cast<Line*>(mouseOverGeometry.at(0));
+    polCoord1.set(1500.f, float(-degree_ + 180));
+    currLine->begin.setFromPolar(polCoord1);
+    currLine->begin += origin;
+    polCoord1.set(1500.f, float(-degree_    ));
+    currLine->end.setFromPolar  (polCoord1);
+    currLine->end   += origin;
+    // update vertical line
+    currLine = static_cast<Line*>(visibleGeometry.at(1));
+    polCoord1.set( 700.f, float(-degree_ + 90 ));
+    currLine->begin.setFromPolar(polCoord1);
+    currLine->begin += origin;
+    polCoord1.set( 700.f, float(-degree_ + 270));
+    currLine->end.setFromPolar  (polCoord1);
+    currLine->end   += origin;
+    currLine = static_cast<Line*>(mouseOverGeometry.at(1));
+    polCoord1.set( 700.f, float(-degree_ + 90 ));
+    currLine->begin.setFromPolar(polCoord1);
+    currLine->begin += origin;
+    polCoord1.set( 700.f, float(-degree_ + 270));
+    currLine->end.setFromPolar  (polCoord1);
+    currLine->end   += origin;
+    // update upper feather line
+    currLine = static_cast<Line*>(visibleGeometry.at(2));
+    polCoord2.set(decay, float(-degree_ + 270));
+    polCoord1.set(350.f, float(-degree_ + 180));
+    currLine->begin.setFromPolar(polCoord1 + polCoord2);
+    currLine->begin += origin;
+    polCoord1.set(350.f, float(-degree_    ));
+    currLine->end.setFromPolar  (polCoord1 + polCoord2);
+    currLine->end   += origin;
+    currLine = static_cast<Line*>(mouseOverGeometry.at(2));
+    polCoord1.set(350.f, float(-degree_ + 180));
+    currLine->begin.setFromPolar(polCoord1 + polCoord2);
+    currLine->begin += origin;
+    polCoord1.set(350.f, float(-degree_    ));
+    currLine->end.setFromPolar  (polCoord1 + polCoord2);
+    currLine->end   += origin;
+    // update lower feather line
+    currLine = static_cast<Line*>(visibleGeometry.at(3));
+    polCoord2.set(decay, float(-degree_ + 90));
+    polCoord1.set(350.f, float(-degree_ + 180));
+    currLine->begin.setFromPolar(polCoord1 + polCoord2);
+    currLine->begin += origin;
+    polCoord1.set(350.f, float(-degree_    ));
+    currLine->end.setFromPolar  (polCoord1 + polCoord2);
+    currLine->end   += origin;
+    currLine = static_cast<Line*>(mouseOverGeometry.at(3));
+    polCoord1.set(350.f, float(-degree_ + 180));
+    currLine->begin.setFromPolar(polCoord1 + polCoord2);
+    currLine->begin += origin;
+    polCoord1.set(350.f, float(-degree_    ));
+    currLine->end.setFromPolar  (polCoord1 + polCoord2);
+    currLine->end   += origin;
+    // update circle's position
+    currCircle = static_cast<Circle*>(visibleGeometry.at(4));
+    currCircle->center = origin;
+    currCircle = static_cast<Circle*>(mouseOverGeometry.at(4));
+    currCircle->center = origin;
```

Comment:
Obviously a lot of duplication here, I am not sure if you prefer it since it is a bit more indirect, but one way to reuse parts of the code is to declare local lambda expressions like
````
    const auto updateLine = [&](Geometry* geometry, const double radius, const double begin, const double end)
    {
        const auto line = static_cast<Line*>(geometry);
        line->begin.setFromPolar(PolarCoord(radius, -degree_ + begin));
        line->begin += origin;
        line->end.setFromPolar(PolarCoord(radius, -degree_ + end));
        line->end += origin;
    };

    const auto updateLineWithDecay = [&](Geometry* geometry, const double radius, const double begin, const double end)
    {
        const auto line = static_cast<Line*>(geometry);
        line->begin.setFromPolar(PolarCoord(radius, -degree_ + begin) + PolarCoord(decay, -degree_ + end));
        line->begin += origin;
        line->end.setFromPolar(PolarCoord(radius, -degree_) + PolarCoord(decay, -degree_ + end));
        line->end += origin;
    };

    const auto updateCircle = [&](Geometry* geometry)
    {
        const auto circle = static_cast<Circle*>(geometry);
        circle->center += origin;
    };

    // update horizontal line
    updateLine (visibleGeometry.at(0), 1500.f, 180.f, 0.f);
    updateLine (mouseOverGeometry.at(0), 1500.f, 180.f, 0.f);
    // update vertical line
    updateLine (visibleGeometry.at(1), 700.f, 90.f, 270.f);
    updateLine (mouseOverGeometry.at(1), 700.f, 90.f, 270.f);
    // update upper feather line
    updateLineWithDecay (visibleGeometry.at(2), 350.f, 180.f, 270.f);
    updateLineWithDecay (mouseOverGeometry.at(2), 350.f, 180.f, 270.f);
    // update lower feather line
    updateLineWithDecay (visibleGeometry.at(3), 350.f, 180.f, 90.f);
    updateLineWithDecay (mouseOverGeometry.at(3), 350.f, 180.f, 90.f);
    // update circle's position
    updateCircle (visibleGeometry.at(4));
    updateCircle (mouseOverGeometry.at(4));
````
and let the compiler take care of inlining them and optimizing away the temporaries. (For this to work, it is necessary to fix `PolarCoord` at least like this
````
diff --git a/rtengine/coord.h b/rtengine/coord.h
index 0b134ce..359fb3a 100644
--- a/rtengine/coord.h
+++ b/rtengine/coord.h
@@ -187,23 +187,23 @@ public:
     {
         radius *= scale;
     }
-    PolarCoord operator+(PolarCoord & rhs)
+    PolarCoord operator+ (const PolarCoord& rhs) const
     {
+        PolarCoord result;
         Coord thisCoord, rhsCoord;
         thisCoord.setFromPolar(*this);
         rhsCoord.setFromPolar(rhs);
         thisCoord += rhsCoord;
-        PolarCoord result;
         result.setFromCartesian(thisCoord);
         return result;
     }
-    PolarCoord operator-(PolarCoord & rhs)
+    PolarCoord operator- (const PolarCoord& rhs) const
     {
+        PolarCoord result;
         Coord thisCoord, rhsCoord;
         thisCoord.setFromPolar(*this);
         rhsCoord.setFromPolar(rhs);
         thisCoord -= rhsCoord;
-        PolarCoord result;
         result.setFromCartesian(thisCoord);
         return result;
     }
```
but that class does not follow best practices in any case for which I'll make a note.)